mountpoint-s3/tests/common/cache.rs

```diff
@@ -0,0 +1,118 @@
+use std::{
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc,
+    },
+    time::Duration,
+};
+
+use async_trait::async_trait;
+use mountpoint_s3::{
+    data_cache::{BlockIndex, ChecksummedBytes, DataCache, DataCacheResult},
+    object::ObjectId,
+};
+
+/// A wrapper around any type implementing [DataCache], which counts operations
+pub struct CacheTestWrapper<Cache> {
+    cache: Arc<Cache>,
+    get_block_ok_count: Arc<AtomicU64>,
+    get_block_hit_count: Arc<AtomicU64>,
+    get_block_failed_count: Arc<AtomicU64>,
+    put_block_ok_count: Arc<AtomicU64>,
+    put_block_failed_count: Arc<AtomicU64>,
+}
```

Comment:
```suggestion
    inner: Arc<CacheTestWrapperInner<Cache>>,
}

struct CacheTestWrapperInner<Cache> {
    cache: Cache,
    get_block_ok_count: AtomicU64,
    get_block_hit_count: AtomicU64,
    get_block_failed_count: AtomicU64,
    put_block_ok_count: AtomicU64,
    put_block_failed_count: AtomicU64,
}
```

Or is there any reason to have all the separate `Arc`s?

Reply:
Also please add some comments on the counters, especially to explain e.g. ok vs hit.